packages/client/test/hooks.spec.ts

```diff
@@ -206,6 +208,48 @@ describe('Hooks', () => {
           ],
         });
       });
+
+      it('"error" must run if resolution details contains an error code', () => {
+        (MOCK_ERROR_PROVIDER.resolveBooleanEvaluation as jest.Mock).mockReturnValue({
+          value: BOOLEAN_VALUE,
+          errorCode: ErrorCode.FLAG_NOT_FOUND,
+        });
+
+        const mockErrorHook = jest.fn();
+
+        const details = client.getBooleanDetails(FLAG_KEY, false, {
+          hooks: [{ error: mockErrorHook }],
+        });
+
+        expect(mockErrorHook).toHaveBeenCalled();
+        expect(details).toEqual(
+          expect.objectContaining({
+            errorCode: ErrorCode.FLAG_NOT_FOUND,
+            reason: StandardResolutionReasons.ERROR,
+          }),
+        );
+      });
+
+      it('"error" must run if resolution details contains the reason "ERROR"', () => {
+        (MOCK_ERROR_PROVIDER.resolveBooleanEvaluation as jest.Mock).mockReturnValue({
+          value: BOOLEAN_VALUE,
+          reason: StandardResolutionReasons.ERROR,
+        });
+
+        const mockErrorHook = jest.fn();
+
+        const details = client.getBooleanDetails(FLAG_KEY, false, {
+          hooks: [{ error: mockErrorHook }],
+        });
+
+        expect(mockErrorHook).toHaveBeenCalled();
+        expect(details).toEqual(
+          expect.objectContaining({
+            errorCode: ErrorCode.GENERAL,
```

Comment:
Error code is automatically set to `GENERAL` if the provider didn't explicitly define a code.